XDG_CACHE_HOME is ignored on macOS

<!--
NOTE: Not filling out the issue template needs a good reason, otherwise it may
take a lot longer to find the problem! Please take the time to help us
debugging the problem by collecting information, even if it seems irrelevant to
you. Thanks!

If you have a question, maybe the forum at https://discourse.restic.net is a
better place.
-->

## Output of `restic version`

restic 0.7.3 (v0.7.3-87-g3afd974d)
compiled with go1.9 on darwin/amd64

## How did you run restic exactly?

<!--
Include the complete command line and any environment variables you used to
configure restic's backend access. Make sure to replace sensitive values!
-->

`restic -r ~/restic-cache-debug backup ~/backup-fodder` 

## What backend/server/service did you use?

Local & B2.

## Expected behavior

If `XDG_CACHE_HOME` is defined, and `--cache-dir` is not specified, then restic should use `$XDG_CACHE_HOME ` for cache.

## Actual behavior

On macOS, restic is ignoring `XDG_CACHE_HOME`, in favor of either or `--cache-dir` flag or `darwinCacheDir()`. From a quick read of the code, Windows has this problem as well.

## Steps to reproduce the behavior

Set XDG_CACHE_HOME to a valid dirname, and run a backup as above.

## Do you have any idea what may have caused this?

I think it was 21338691276a138f6e0bb2ead9068683c38ddf9f.

## Do you have an idea how to solve the issue?

If `XDG_CACHE_HOME` is only meant to be supported on Linux (perhaps reasonable to do so?) then the documentation should reflect that. Else, maybe [something like this](https://github.com/armhold/restic/commit/8968fd7fdfd3625d77bca05903e15e004b12a9b0) (I have not tested that code.)

PS: how can I run a specific unit test file? I am not sure what is tripping up the package resolution here. Also tried running the test from the package dir. 

```
go test ./internal/cache/file_test.go 
# command-line-arguments
internal/cache/file_test.go:16:64: undefined: Cache
internal/cache/file_test.go:44:28: undefined: Cache
internal/cache/file_test.go:66:33: undefined: Cache
internal/cache/file_test.go:75:34: undefined: Cache
internal/cache/file_test.go:86:16: undefined: TestNewCache
internal/cache/file_test.go:139:16: undefined: TestNewCache
internal/cache/file_test.go:197:16: undefined: TestNewCache
FAIL	command-line-arguments [build failed]
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,10 +7,10 @@ Important Changes in 0.X.Y
  * We've added a local cache for metadata so that restic doesn't need to load
    all metadata (snapshots, indexes, ...) from the repo each time it starts. By
    default the cache is active, but there's a new global option `--no-cache`
-   that can be used to disable the cache. The cache location is
-   `~/.cache/restic` by default, which can be overridden with `--cache-dir` or
-   the environment variable `XDG_CACHE_HOME`. The cache will automatically
-   populate, indexes and snapshots are saved as they are loaded.
+   that can be used to disable the cache. By deafult, the cache a standard
+   cache folder for the OS, which can be overridden with `--cache-dir`.  The
+   cache will automatically populate, indexes and snapshots are saved as they
+   are loaded.
    https://github.com/restic/restic/pull/1040
    https://github.com/restic/restic/issues/29
    https://github.com/restic/restic/issues/738

--- a/doc/manual_rest.rst
+++ b/doc/manual_rest.rst
@@ -267,12 +267,16 @@ Caching
 
 Restic keeps a cache with some files from the repository on the local machine.
 This allows faster operations, since meta data does not need to be loaded from
-a remote repository. The cache is automatically created, usually in the
-directory ``.cache/restic`` in the user's home directory. The environment
-variable ``XDG_CACHE_DIR`` or the command line parameter ``--cache-dir`` can
-each be used to specify where the cache is located. The parameter
-``--no-cache`` disables the cache entirely. In this case, all data is loaded
-from the repo.
+a remote repository. The cache is automatically created, usually in an
+OS-specific cache folder:
+
+ * Linux/other: ``~/.cache/restic`` (or ``$XDG_CACHE_HOME/restic``)
+ * macOS: ``~/Library/Caches/restic``
+ * Windows: ``%LOCALAPPDATA%/restic``
+
+The command line parameter ``--cache-dir`` can each be used to override the
+default cache location. The parameter ``--no-cache`` disables the cache
+entirely. In this case, all data is loaded from the repo.
 
 The cache is ephemeral: When a file cannot be read from the cache, it is loaded
 from the repository.

